source/ranges.tex

```diff
@@ -9341,6 +9357,550 @@
+    constexpr @\exposid{iterator}@ operator++(int) requires @\libconcept{forward_range}@<@\exposid{Base}@>;
+
+    constexpr @\exposid{iterator}@& operator--() requires @\libconcept{bidirectional_range}@<@\exposid{Base}@>;
+    constexpr @\exposid{iterator}@ operator--(int) requires @\libconcept{bidirectional_range}@<@\exposid{Base}@>;
+
+    constexpr @\exposid{iterator}@& operator+=(difference_type x)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+    constexpr @\exposid{iterator}@& operator-=(difference_type x)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+
+    constexpr auto operator[](difference_type n) const
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>
+      { return @\exposid{reference-type}@(@\exposid{pos_}@ + n, @\exposid{current_}@[n]); }
+
+    friend constexpr bool operator==(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y) noexcept;
+    friend constexpr strong_ordering operator<=>(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y) noexcept;
+    friend constexpr @\exposid{iterator}@ operator+(const @\exposid{iterator}@& x, difference_type y)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+    friend constexpr @\exposid{iterator}@ operator+(difference_type x, const @\exposid{iterator}@& y)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+    friend constexpr @\exposid{iterator}@ operator-(const @\exposid{iterator}@& x, difference_type y)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+    friend constexpr difference_type operator-(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y);
+
+    friend constexpr auto iter_move(const @\exposid{iterator}@& i)
+      noexcept(noexcept(ranges::iter_move(i.@\exposid{current_}@)) &&
+               is_nothrow_move_constructible_v<range_rvalue_reference_t<@\exposid{Base}@>>) {
+      return tuple<difference_type,
+                   range_rvalue_reference_t<@\exposid{Base}@>>(@\exposid{pos_}@, ranges::iter_move(i.@\exposid{current_}@));
+    }
+  };
+}
+\end{codeblock}
+
+\pnum
+The member \grammarterm{typedef-name}
+\tcode{\exposid{iterator}::iterator_concept}
+is defined as follows:
+\begin{itemize}
+\item
+If \exposid{Base} models \libconcept{random_access_range},
+then \tcode{iterator_concept} denotes \tcode{random_access_iterator_tag}.
+\item
+Otherwise, if \exposid{Base} models \libconcept{bidirectional_range},
+then \tcode{iterator_concept} denotes \tcode{bidirectional_iterator_tag}.
+\item
+Otherwise, if \exposid{Base} models \libconcept{forward_range},
+then \tcode{iterator_concept} denotes \tcode{forward_iterator_tag}.
+\item
+Otherwise, \tcode{iterator_concept} denotes \tcode{input_iterator_tag}.
+\end{itemize}
+
+\indexlibraryctor{enumerate_view::\exposid{iterator}}%
+\begin{itemdecl}
+constexpr explicit @\exposid{iterator}@(iterator_t<@\exposid{Base}@> current, difference_type pos = 0);
```

Comment:
It's certainly not present in the class synopsis. It the implementation doesn't make use of it, it must be an error here. If it does, it's an error in the class synopsis.

Reply:
@cor3ntin, @CaseyCarter: is this an error in the paper? Should there be a default argument in the member declaration of this constructor, or should there just not be any default argument?

Reply:
It would appear so;  I think there should not be any default argument at all given that it's never used ( `enumerate_view::begin` passes `0` explicitly when constructing the iterator)

Reply:
Why is the ctor `explicit`?

Reply:
@cor3ntin ^^^ ?

Reply:
It will be LWG 3908

Reply:
LWG3908